### PR TITLE
Feat error logging

### PR DESCRIPTION
**Pull request by jmpearson135**
Added a function to call for logging errors the format is logIT('errorType', 'error', '__LINE__', '__FILE__').
Error type can be 'db', 'frontend',  'api'. 
Error what ever is caught with the try catch. 
__LINE__ and __FILE__ put in the function exactly like that and will automatically grab the line # and file name. 
image

Added code for four logServers for each virtual machine so that the logs can be grabbed from each persons rabbit queue and saved to their individual vm so all logs are shared. 
image

Modified config.php files for new rabbitmq exchanges and queues for logging. 
image

Added new server information to rabbitMQ.ini file.
image

Added logIT function to request handling pages in catch statements. 

image
image
image

### CODE REVIEW

File: data/configuration/config.php [hunk 1]
```diff
@@ -9,8 +9,12 @@
 // Rabbit MQ Configs
```
Comment (Carlomos7) [inline data/configuration/config.php:10]: you can remove this since it's set in data/RabbitMQ/rbMQconfig.php:

File: scripts/rabbitMQserver.php [hunk 2]
```diff
@@ -20,7 +20,7 @@ function requestProcessor($request)
             $response = handle_login($request["username"],$request["password"]);
             break;
         case "register":
-            $response = handle_register($request["email"],$request["username"],$request["password"]);
+            $response = handle_register($request["username"],$request["password"]);
```
Comment (Carlomos7) [inline scripts/rabbitMQserver.php:23]: This change overrides the email value provided

File: src/lib/config.php [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <?php
```
Comment (Carlomos7) [inline src/lib/config.php:1]: Remove to not override config.php

File: src/lib/jwt_handling/generate_jwt.php [hunk 4]
```diff
@@ -7,9 +7,7 @@
 
 function generate_jwt($db, $user){
```
Comment (Carlomos7) [inline src/lib/jwt_handling/generate_jwt.php:8]: Merge Conflict

File: src/lib/jwt_handling/validate_jwt.php [hunk 5]
```diff
@@ -9,9 +9,7 @@
 function validate_jwt($jwt) {
     $key = JWT_SECRET;
```
Comment (Carlomos7) [inline src/lib/jwt_handling/validate_jwt.php:10]: Merge conflict

File: src/lib/request_handling/handle_register.php [hunk 6]
```diff
@@ -1,13 +1,11 @@
 <?php
```
Comment (Carlomos7) [inline src/lib/request_handling/handle_register.php:1]: Merge Conflict

File: src/lib/request_handling/handle_logout.php [hunk 7]
```diff
@@ -4,9 +4,7 @@
 
 function handle_logout($jwt){
```
Comment (Carlomos7) [inline src/lib/request_handling/handle_logout.php:5]: Merge conflict

File: src/lib/request_handling/handle_login.php [hunk 8]
```diff
@@ -3,18 +3,15 @@
 
 function handle_login($username,$password){
         $db = getDB();
-        $table_name = 'Users';
-        $query = "SELECT id, username, password FROM $table_name WHERE username = :username";
-        $stmt = $db->prepare($query);
+        $stmt = $db->prepare("SELECT id, username, password FROM testusers WHERE username = :username");
```
Comment (Carlomos7) [inline src/lib/request_handling/handle_login.php:6]: Merge Conflict

**Review by Carlomos7 — CHANGES_REQUESTED** [review]

**Review by Carlomos7 — CHANGES_REQUESTED** [review]

File: scripts/jaylinErrorLogging.php [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+             logError($error, '/var/www/sample/data/logs/frontendErrors');
+             break;
+                       
+        case "db":
+            //uses logError function to write the errors to the file dbErrors
+             logError($error, '/var/www/sample/data/logs/dbErrors');
+             break;
+        case "api":
+            //uses logError function to write the errors to the file apiErrors    
+             logError($error, '/var/www/sample/data/logs/apiErrors');
```
[the remark this answers is not in the document]
Comment (jmpearson135) [reply scripts/jaylinErrorLogging.php]: Fixed the directory paths in the logging files and changed the apiErrors file name. 
https://github.com/AudioNook/IT490-002/pull/105/commits/590387c069b37a9cf1f42f5030cdefa84bf1d91d
https://github.com/AudioNook/IT490-002/pull/105/commits/2f9b256956c869c1f8e3f21f6bfd35344cd35426
https://github.com/AudioNook/IT490-002/pull/105/commits/20511d59365e8dbe045c498e831e499939bbea20
https://github.com/AudioNook/IT490-002/pull/105/commits/7ac5e5d34e947b82649be8d2673a826edda51ab4
https://github.com/AudioNook/IT490-002/pull/105/commits/baea5abd1d0b6fb77caa6f1103475045825abc67
https://github.com/AudioNook/IT490-002/pull/105/commits/8083a542ac2b562408d32f65d2a98f9d286f42b7
https://github.com/AudioNook/IT490-002/pull/105/commits/0ff297b66bfc44bd17d954813f71ba6131bb600a
https://github.com/AudioNook/IT490-002/pull/105/commits/69654771564d03f535b26bab98b45b47bbf4b705

**Review by Carlomos7 — APPROVED** [review]

**Pull request merged**